src/sentry/buffer/redis.py

```diff
@@ -82,6 +84,16 @@ def _make_key(self, model, filters):
             ).hexdigest(),
         )
 
+    def _make_pending_key(self, shard=None):
```

Comment:
Why support the `None` case separately than just a 0 shard? Is this for compatibility or something?

Reply:
OK, I read this below, I think it'd be helpful to write a docstring about how this is structured now but I'm not going to block on it. It might be (almost certainly is) worth doing either `assert shard != 0 or self.pending_shards > 1` or adding a similar expression to this conditional so you can't accidentally `_make_pending_key` with shard 0 that is never processed as part of `process_pending`.

Reply:
Yeah, `None` is simply the case where there is no sharding. In which case, you're right, I could add some asserts. I'll add to my follow up as well for that. And some docstrings.